Allow running on windows

This extension currently cannot be used on windows due to issues with the path (`Error: Cannot find module 'C:\C:\Users\`). The suggested fix is https://github.com/zed-industries/zed/issues/20559#issuecomment-2755397781

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+# 1.0.1 - 2025-04-04
+
+- Make it work on windows
+
 # 1.0.0 - 2024-08-02
 
 - Add GraphQL language server with auto-completion and documentation support. Thanks @affanshahid

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "zed_graphql"
-version = "0.0.3"
+version = "1.0.1"
 edition = "2021"
 publish = false
 

--- a/extension.toml
+++ b/extension.toml
@@ -1,6 +1,6 @@
 id = "graphql"
 name = "GraphQL"
-version = "1.0.0"
+version = "1.0.1"
 schema_version = 1
 authors = ["Ivan Buryak <ivan.buryak@gmail.com>"]
 description = "GraphQL extension for Zed"

--- a/src/graphql.rs
+++ b/src/graphql.rs
@@ -73,8 +73,7 @@ impl zed::Extension for GraphQLExtension {
         Ok(zed::Command {
             command: zed::node_binary_path()?,
             args: vec![
-                env::current_dir()
-                    .unwrap()
+                zed_ext::sanitize_windows_path(env::current_dir().unwrap())
                     .join(&server_path)
                     .to_string_lossy()
                     .to_string(),
@@ -90,3 +89,25 @@ impl zed::Extension for GraphQLExtension {
 }
 
 zed::register_extension!(GraphQLExtension);
+
+/// Extensions to the Zed extension API that have not yet stabilized.
+mod zed_ext {
+    /// Sanitizes the given path to remove the leading `/` on Windows.
+    ///
+    /// On macOS and Linux this is a no-op.
+    ///
+    /// This is a workaround for https://github.com/bytecodealliance/wasmtime/issues/10415.
+    pub fn sanitize_windows_path(path: std::path::PathBuf) -> std::path::PathBuf {
+        use zed_extension_api::{current_platform, Os};
+
+        let (os, _arch) = current_platform();
+        match os {
+            Os::Mac | Os::Linux => path,
+            Os::Windows => path
+                .to_string_lossy()
+                .to_string()
+                .trim_start_matches('/')
+                .into(),
+        }
+    }
+}

